lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs

```diff
@@ -2587,22 +2594,282 @@ spec = describe "SHARED_TRANSACTIONS" $ do
+            verify rl [expectListSize 10]
+        rl1b <- request @([ApiTransaction n]) ctx linkList1a Default Empty
+        verify rl1b [expectListSize 1]
+
+        -- next the funded wallet sends back funds in such a way that we can
+        -- indetify address 1 was engaged in a given tx
+        let a5 = fromIntegral $ oneAda * 4_990
+
+        sendAmtToAddr ctx wDest wSrc a5 0
+        eventually "There are exactly 11 transactions for wDest" $ do
+            let linkList = Link.listTransactions' @'Shared wDest
+                    Nothing
+                    Nothing
+                    Nothing
+                    Nothing
+                    Nothing
+                    Nothing
+            rl <- request @([ApiTransaction n]) ctx linkList Default Empty
+            verify rl [expectListSize 11]
+
+        rl1c <- request @([ApiTransaction n]) ctx linkList1a Default Empty
+        verify rl1c [expectListSize 2]
   where
+     oneAda :: Integer
+     oneAda = 1_000_000
+
+     listExternalAddresses ctx w = do
+         let link = Link.listAddresses @'Shared w
+         r <- request @[ApiAddressWithPath n] ctx link Default Empty
+         expectResponseCode HTTP.status200 r
+         let isExternal (ApiAddressWithPath _ _
+                 (_ NE.:| [_, _, (ApiT (DerivationIndex ix)), _] ) ) = ix == 0
+             isExternal _ = False
+         return (filter isExternal $ getFromResponse Prelude.id r)
+
+     sendAmtToAddr ctx src dest amt addrIx = do
+         (_, Right (ApiSharedWallet (Right wal1))) <-
+             getSharedWallet ctx (ApiSharedWallet (Right dest))
+         let bal1 = wal1 ^. (#balance . #available)
+
+         (_, Right (ApiSharedWallet (Right wal2))) <-
+             getSharedWallet ctx (ApiSharedWallet (Right src))
+         let bal2 = wal2 ^. (#balance . #available)
+
+         addrs <- listExternalAddresses ctx dest
+         let destination = (addrs !! addrIx) ^. #id
+         let payload = Json [json|{
+                 "payments": [{
+                     "address": #{destination},
+                     "amount": {
+                         "quantity": #{amt},
+                         "unit": "lovelace"
+                     }
+                 }]
+             }|]
+
+         rTx <- request @(ApiConstructTransaction n) ctx
+             (Link.createUnsignedTransaction @'Shared src) Default payload
+         verify rTx
+             [ expectResponseCode HTTP.status202
+             ]
+
+         let (ApiSerialisedTransaction apiTx _) =
+                 getFromResponse #transaction rTx
+         signedTx <-
+             signSharedTx ctx src apiTx [ expectResponseCode HTTP.status202 ]
+
+         submittedTx <- submitSharedTxWithWid ctx src signedTx
+         verify submittedTx
+             [ expectSuccess
+             , expectResponseCode HTTP.status202
+             ]
+         eventually "dest wallet balance is higher than before" $ do
+             rGet <- request @ApiWallet ctx
```

Comment:
👍 This is necessary to make the test robust for execution on an actual Cardano network, where concurrency, consensus, and timing can and will affect the results — typically leading to "flaky" test failure.

However, instead of checking the balances, I prefer checking that for each wallet, the transaction (identified by its transaction ID) has status `in_ledger` — this is the correct way to ensure that a transaction has been accommodated in both source and destination wallet.

Reply:
very reasonable idea! Did it for all styles - byron, shared and shelley!